mcpgateway/plugins/framework/external/mcp/server/server.py

```diff
@@ -117,36 +118,42 @@ async def invoke_hook(
             >>> import asyncio
             >>> import os
             >>> os.environ["PYTHONPATH"] = "."
-            >>> from mcpgateway.plugins.framework import PromptPrehookPayload, PluginContext, PromptPrehookResult
+            >>> from mcpgateway.plugins.framework import GlobalContext, PromptPrehookPayload, PluginContext, PromptPrehookResult
             >>> server = ExternalPluginServer(config_path="./tests/unit/mcpgateway/plugins/fixtures/configs/valid_multiple_plugins_filter.yaml")
             >>> def prompt_pre_fetch_func(plugin: Plugin, payload: PromptPrehookPayload, context: PluginContext) -> PromptPrehookResult:
             ...     return plugin.prompt_pre_fetch(payload, context)
             >>> payload = PromptPrehookPayload(name="test_prompt", args={"user": "This is so innovative"})
-            >>> context = PluginContext(request_id="1", server_id="2")
+            >>> context = PluginContext(global_context=GlobalContext(request_id="1", server_id="2"))
             >>> initialized = asyncio.run(server.initialize())
             >>> initialized
             True
             >>> result = asyncio.run(server.invoke_hook(PromptPrehookPayload, prompt_pre_fetch_func, "DenyListPlugin", payload.model_dump(), context.model_dump()))
             >>> result is not None
             True
-            >>> result["continue_processing"]
+            >>> result["result"]["continue_processing"]
             False
         """
         global_plugin_manager = PluginManager()
         plugin_timeout = global_plugin_manager.config.plugin_settings.plugin_timeout if global_plugin_manager.config else DEFAULT_PLUGIN_TIMEOUT
         plugin = global_plugin_manager.get_plugin(plugin_name)
+        result_payload: dict[str, Any] = {}
```

Comment:
How about we do this?
```python
- result_payload: dict[str, Any] = {}
+ result_payload: dict[str, Any] = {"plugin": plugin_name}
```
This would always return the plugin_name in the result, which could be handy for logging and other things without extra lookups.

Reply:
done!